test.sh

```diff
@@ -1,80 +1,129 @@
-            exit 1
-        fi
-        export STUB_UNIVERSE_URL=$(cat $UNIVERSE_URL_PATH)
-        rm -f $UNIVERSE_URL_PATH
-        echo "Built/uploaded stub universe: $STUB_UNIVERSE_URL"
-    else
-        echo "Using provided STUB_UNIVERSE_URL: $STUB_UNIVERSE_URL"
-    fi
+if [ "$#" -eq "0" ]; then
+    usage
+fi
 
-    echo Security: $SECURITY
-    if [ "$SECURITY" = "strict" ]; then
-        ${REPO_ROOT_DIR}/tools/setup_permissions.sh root ${framework}-role
-        # include foldered roles (tests exercise with /test/integration/svcname):
-        ${REPO_ROOT_DIR}/tools/setup_permissions.sh root test__integration__${framework}-role
-    fi
+if [ -z "$CLUSTER_URL" ]; then
+    echo "Cluster not found. Create and configure one then set \$CLUSTER_URL."
+    exit 1
+fi
 
-    # Run shakedown tests in framework directory:
-    TEST_GITHUB_LABEL="${framework}" ${REPO_ROOT_DIR}/tools/run_tests.py shakedown ${FRAMEWORK_DIR}/tests/
-}
+if [ -z "$AWS_ACCESS_KEY_ID" -o -z "$AWS_SECRET_ACCESS_KEY" ]; then
+    echo "AWS credentials not found (\$AWS_ACCESS_KEY_ID and \$AWS_SECRET_ACCESS_KEY)."
+    exit 1
+fi
 
-echo "Beginning integration tests at "`date`
+security="permissive"
+pytest_m="sanity and not azure"
+pytest_k=""
+azure_args=""
+ssh_path="${HOME}/.ssh/ccm.pem"
 
-REPO_ROOT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+# If AZURE variables are given, change default -m and prepare args for docker
+if [ -n "$AZURE_DEV_CLIENT_ID" -a -n "$AZURE_DEV_CLIENT_SECRET" -a \
+        -n "$AZURE_DEV_TENANT_ID" -a -n "$AZURE_DEV_STORAGE_ACCOUNT" -a \
+        -n "$AZURE_DEV_STORAGE_KEY" ]; then
+azure_args=$(cat<<EOFF
+    -e AZURE_CLIENT_ID=$AZURE_DEV_CLIENT_ID \
+    -e AZURE_CLIENT_SECRET=$AZURE_DEV_CLIENT_SECRET \
+    -e AZURE_TENANT_ID=$AZURE_DEV_TENANT_ID \
+    -e AZURE_STORAGE_ACCOUNT=$AZURE_DEV_STORAGE_ACCOUNT \
+    -e AZURE_STORAGE_KEY=$AZURE_DEV_STORAGE_KEY
+EOFF
+)
+    pytest_m="sanity"
+fi
 
-cd $REPO_ROOT_DIR
+while [[ $# -gt 1 ]]
+do
+key="$1"
 
-# Get a CCM cluster if not already configured (see available settings in dcos-commons/tools/README.md):
-if [ -z "$CLUSTER_URL" ]; then
-    echo "CLUSTER_URL is empty/unset, launching new cluster."
-    export CCM_AGENTS=6
-    CLUSTER_INFO=$(${REPO_ROOT_DIR}/tools/launch_ccm_cluster.py)
-    echo "Launched cluster: ${CLUSTER_INFO}"
-    # jq emits json strings by default: "value".  Use --raw-output to get value without quotes
-    export CLUSTER_URL=$(echo "${CLUSTER_INFO}" | jq --raw-output .url)
-    export CLUSTER_ID=$(echo "${CLUSTER_INFO}" | jq .id)
-    export CLUSTER_AUTH_TOKEN=$(echo "${CLUSTER_INFO}" | jq --raw-output .auth_token)
-    CLUSTER_CREATED="true"
-else
-    echo "Using provided CLUSTER_URL as cluster: $CLUSTER_URL"
-    CLUSTER_CREATED=""
-fi
+case $key in
+    -m)
+    pytest_m="$2"
+    shift # past argument
+    ;;
+    -k)
+    pytest_k="$2"
+    shift # past argument
+    ;;
+    -s)
+    security="strict"
+    shift # past argument
```

Comment:
(No problem with regards to the WIP. Was just looking through things, and noticed that. Thought it may cause some unexpected behaviour).